tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -2168,6 +2168,39 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  /** Translate litterals to there ScalaJS String representation. */
+  private def literalToString(l: Literal): Option[String] = l match {
+    case StringLiteral(value)  => Some(value)
+    case FloatLiteral(value)   => nuStr(value.toDouble)
+    case DoubleLiteral(value)  => nuStr(value)
+    case LongLiteral(value)    => Some(value.toString)
+    case IntLiteral(value)     => Some(value.toString)
+    case BooleanLiteral(value) => Some(value.toString)
+    case Null()                => Some("null")
+    case Undefined()           => Some("undefined")
+
+    /* It is not possible to constant fold the ClassOf case without
+     * knowing wether classOf[T] represent a class or an interface and
+     * there are no possibilites to get that information in the optimizer.
+     */
+    case ClassOf(_) => None
+  }
+
+  /* The JVM and ECMAScript 6 specification do not match
+   * for float and double, as such no constant folding is done on them.
+   */
+  private def nuStr(value: Double): Option[String] = {
+    if(value.isValidInt) {
+      Some(if (value == 0.0) "0" // for the -0.0 case.
+           else value.toString.stripSuffix(".0"))
```

Comment:
Technically, there's an indentation issue, here (5 spaces). I don't know how to fix it without making it uglier, though. That's funny.